### PR TITLE
dynamic host volumes: test client RPC and plugins

### PR DESCRIPTION
**Pull request by gulducat**
adding tests for #24497 

also happened to notice we weren't validating that volume ID was uuid-shaped, so add that too.

main issue: #24479

### CODE REVIEW

File: nomad/structs/host_volumes.go [hunk 1]
```diff
@@ -167,7 +171,7 @@ func (hv *HostVolume) Validate() error {
 		}
 	}
 
-	return mErr.ErrorOrNil()
+	return helper.FlattenMultierror(mErr.ErrorOrNil())
```
Comment (tgross) [inline nomad/structs/host_volumes.go:174]: Nice QoL catch.

File: client/hostvolumemanager/host_volume_plugin_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+func TestHostVolumePluginMkdir(t *testing.T) {
+	volID := "test-vol-id"
+	tmp := t.TempDir()
+	target := filepath.Join(tmp, volID)
+
+	plug := &HostVolumePluginMkdir{
+		ID:         "test-mkdir-plugin",
+		TargetPath: tmp,
+		log:        hclog.Default(),
```
Comment (tgross) [inline client/hostvolumemanager/host_volume_plugin_test.go:30]: There's a `helper/testlog` package that has some nice features like only emitting logs on failure (or `go test -v`) and including the location in the log outputs:

```suggestion
		log:        testlog.Logger(t),
```

Unfortunately it doesn't have a recorder, which would have been handy for your other test below.

File: client/hostvolumemanager/host_volume_plugin_test.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+)
+
+func TestHostVolumePluginMkdir(t *testing.T) {
+	volID := "test-vol-id"
+	tmp := t.TempDir()
+	target := filepath.Join(tmp, volID)
+
+	plug := &HostVolumePluginMkdir{
+		ID:         "test-mkdir-plugin",
+		TargetPath: tmp,
```
Comment (tgross) [inline client/hostvolumemanager/host_volume_plugin_test.go:29]: This makes me realize that calling this field "target path" is maybe a little misleading, as it's the parent of the eventual target path.

**Review by tgross — APPROVED** [review]
LGTM!

File: client/hostvolumemanager/host_volume_plugin_test.go (hunk 2 above)
Comment (gulducat) [reply client/hostvolumemanager/host_volume_plugin_test.go:30]: looks like I need a `testlog.HCLogger(t)` here, but I like!

File: client/hostvolumemanager/host_volume_plugin_test.go (hunk 3 above)
Comment (gulducat) [reply client/hostvolumemanager/host_volume_plugin_test.go:29]: yeah, aside from "target", even "path" is ambiguous: could signify either a file or a dir.  I'll be in this vicinity again with fingerprinting, so I'll keep this in mind.

**Pull request merged**